model/src/test/kotlin/licenses/LicenseConfigurationTest.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.licenses
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.containExactly
+import io.kotest.matchers.collections.containExactlyInAnyOrder
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+
+import java.lang.IllegalStateException
+import java.util.Collections.emptySortedSet
+
+import org.ossreviewtoolkit.spdx.SpdxSingleLicenseExpression
+
+class LicenseConfigurationTest : WordSpec({
+    "LicenseConfiguration.init" should {
+        "detect duplicate category names" {
+            val cat1 = LicenseCategory("Category 1")
+            val cat2 = LicenseCategory("Category 2", "Another category")
+            val cat3 = LicenseCategory("Category 1", "Duplicate; should cause a failure")
+
+            shouldThrow<IllegalArgumentException> {
+                LicenseConfiguration(categories = listOf(cat1, cat2, cat3))
+            }
+        }
+
+        "detect duplicate license IDs" {
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), emptySortedSet()
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), emptySortedSet()
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf("permissive")
+            )
+
+            shouldThrow<IllegalArgumentException> {
+                LicenseConfiguration(categorizations = listOf(lic1, lic2, lic3))
+            }
+        }
+
+        "detect licenses referencing non-existing categories" {
+            val cat1 = LicenseCategory("Category 1")
+            val cat2 = LicenseCategory("Category 2")
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf(cat1.name)
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), sortedSetOf("unknownCategory")
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("BSD"), sortedSetOf("anotherUnknownCategory")
+            )
+
+            val exception = shouldThrow<java.lang.IllegalArgumentException> {
```

Comment:
Please just use `IllegalArgumentException` here; no import should be required (see existing code).